app/scripts/modules/netflix/pipeline/stage/isolatedTestingTarget/isolatedTestingTargetStage.ts

```diff
@@ -0,0 +1,252 @@
+    return vips ? [vips.oldVip, vips.oldSecureVip].filter(n => n).join(', ') : '';
+  };
+
+  public getClusterNewVIPs(cluster: any): string {
+    const vips = this.stage.vipOverrides[this.getClusterId(cluster)];
+    return vips ? [vips.newVip, vips.newSecureVip].filter(n => n).join(', ') : '';
+  };
+
+  public getClusterName(cluster: any): string {
+    return this.namingService.getClusterName(cluster.application, cluster.stack, cluster.freeFormDetails);
+  }
+
+  private buildServerGroupCommand(selectedProvider: string): ng.IPromise<any> {
+  return this.serverGroupCommandBuilder.buildNewServerGroupCommandForPipeline(selectedProvider)
+    .then((command: any) => {
+      this.configureServerGroupCommandForEditing(command);
+      command.viewState.overrides = {
+        useSourceCapacity: false,
+        capacity: {
+          min: 1, max: 1, desired: 1,
+        },
+        loadBalancers: [],
+        freeFormDetails: 'itt-' + this.stage.username,
+        maxRemainingAsgs: 2
+      };
+      command.viewState.disableNoTemplateSelection = true;
+      command.viewState.customTemplateMessage = 'Select a template to configure the isolated testing target cluster. ' +
+        'If you want to configure the server groups differently, you can do so by clicking ' +
+        '"Edit" after adding the cluster.';
+      return command;
+    });
+  }
+
+  private applyCommandToStage(command: any): void {
+    // Push the cluster config into the list of clusters
+    const cluster = this.awsServerGroupTransformer.convertServerGroupCommandToDeployConfiguration(command);
+    this.stage.clusters.push(cluster);
+
+    // If the ASG we are cloning has a VIP, generate the new VIP
+    if (command.source && command.source.asgName && command.source.region && command.source.account) {
+      this.serverGroupReader.getServerGroup(command.application, command.source.account, command.source.region, command.source.asgName).then((asgDetails: ServerGroup) => {
+        if (asgDetails) {
+          const discoveryHealth = filter(flatten(map(asgDetails.instances, 'health')), { type: 'Discovery' });
+
+          let oldVip: string, oldSecureVip: string;
+          each(discoveryHealth, (health: any) => {
+            oldVip = oldVip || health.vipAddress;
+            oldSecureVip = oldSecureVip || health.secureVipAddress;
+          });
+
+          const newVip = this.generateNewVip(oldVip, command.freeFormDetails);
+          const newSecureVip = this.generateNewVip(oldSecureVip, command.freeFormDetails);
+          this.stage.vipOverrides[this.getClusterId(cluster)] = { oldVip, oldSecureVip, newVip, newSecureVip };
+        }
+      });
+    }
+  }
+
+  public addCluster(): void {
+    this.stage.clusters = this.stage.clusters || [];
+    this.stage.vipOverrides = this.stage.vipOverrides || {};
+    this.providerSelectionService.selectProvider(this.$scope.application).then((selectedProvider: string) => {
```

Comment:
If this is only for AWS, we can omit this bit - but if it's designed to work with Titus, too, we should keep it in here.